code/modules/power/apc.dm

```diff
@@ -726,6 +757,21 @@
 				spark_system.queue()
 				H << "<span class='danger'>The APC power currents surge eratically, damaging your chassis!</span>"
 				H.adjustFireLoss(10, 0)
+			if(infected)
+				if(H in hacked_ipcs)

```

Comment:
This won't work. You should do `if ("\ref[H]" in hacked_ipcs)`.

Because lazylist is used, for safety: `if (LAZYLEN(hacked_ipcs) && ("\ref[H]" in hacked_ipcs))`

I'm unsure how BYOND will handle `if (x in null)`. Maybe @Lohikar can clarify.